darts/tests/optional_deps/test_optuna.py

```diff
@@ -0,0 +1,180 @@
+import os
+from itertools import product
+
+import numpy as np
+import pytest
+from sklearn.preprocessing import MaxAbsScaler
+
+from darts.dataprocessing.transformers import Scaler
+from darts.datasets import AirPassengersDataset
+from darts.metrics import smape
+from darts.models import LinearRegressionModel
+from darts.tests.conftest import OPTUNA_AVAILABLE, TORCH_AVAILABLE, tfm_kwargs
+
+if not OPTUNA_AVAILABLE:
+    pytest.skip(
+        f"Optuna not available. {__name__} tests will be skipped.",
+        allow_module_level=True,
+    )
+
+import optuna
+
+if TORCH_AVAILABLE:
+    import torch
+    from pytorch_lightning.callbacks import Callback, EarlyStopping
+
+    # hacky workaround found in https://github.com/Lightning-AI/pytorch-lightning/issues/17485
+    # to avoid import of both lightning and pytorch_lightning
+    class PatchedCallback(optuna.integration.PyTorchLightningPruningCallback, Callback):
```

Comment:
does this work now? If so then we should probably update the example in the user guide